lib/module.h

```diff
@@ -8,7 +8,24 @@
 #include <cstdint>
 
 namespace module {
-  uintptr_t findModule(const char* moduleName, pid_t processId, const char** errorMessage);
-};
+
+    struct Module {
+        uintptr_t start;
+        uintptr_t end;
+        char permissions[32];
+        unsigned long long offset;
+        unsigned dev_major;
+        unsigned dev_minor;
+        unsigned long long inode;
+        char *pathname;
+
+        ~Module() {
+            delete pathname;
```

Comment:
Gah I hate to be a stickler, but `pathname` is allocated with `malloc` via `strdup` -- so it needs to be [deallocated with `free(...)` not `delete`](https://isocpp.org/wiki/faq/freestore-mgmt#mixing-malloc-and-delete).

(I'm probably partially to blame here -- I'm not as familiar with C++ as C, so I leaned a lot on C idioms in this C++ code...)

Reply:
No worries, I come from the OOP world myself with everything cleared for me. Only recently started getting into C/C++ and the need to free my own variables :D